### ISSUE TITLE
Problems with docker scripts

### ISSUE THREAD

**Issue by blattms** [OPM/opm-reference-manual#213]
I am not sure what exactly is happening, but I cannot open the files using the docker scripts.  It seems like they are not available there.

I I start the container and try to open main.fodt I get:
```
(venv) mblatt@frau-mahlzahn:~/opm-reference-manual$ lodocker-open-file-and-update parts/main.fodt 
INFO:werkzeug:172.17.0.1 - - [18/Mar/2024 08:50:11] "POST /open-document-and-update HTTP/1.1" 404 -
Failed to open document parts/main.fodt. Status code: 404
(venv) mblatt@frau-mahlzahn:~/opm-reference-manual$ 
```

Before that I tried to do this with the script `lodocker-run-container parts/main.fodt` but the file is not found.

**Comment by hakonhagland** [comment]
@blattms Did you try with `lodocker-run-container main.fodt` ? The filename is relative to the `parts` folder

**Comment by hakonhagland** [comment]
The docker-container just mounts the `parts` from the host into `/home/docker-user/parts` in the container. So the container will not now what is the user's current directory on the host when he runs `lodocker-run-container`. That is why it made sense to have the filename be relative to `parts` in the container. @blattms Should we change this behavior? For example, if the user run `lodocker-run-container parts/main.fodt` from the root of the repository, the script could remove the `parts` when submitting the path to the container

**Comment by blattms** [comment]
Yes:
```sh
(venv) mblatt@frau-mahlzahn:~/opm-reference-manual$ lodocker-open-file-and-update main.fodt 
INFO:werkzeug:172.17.0.1 - - [18/Mar/2024 09:55:27] "POST /open-document-and-update HTTP/1.1" 404 -
Failed to open document main.fodt. Status code: 404
(venv) mblatt@frau-mahlzahn:~/opm-reference-manual$ cd parts/
(venv) mblatt@frau-mahlzahn:~/opm-reference-manual/parts$ lodocker-open-file-and-update main.fodt 
INFO:werkzeug:172.17.0.1 - - [18/Mar/2024 09:55:38] "POST /open-document-and-update HTTP/1.1" 404 -
Failed to open document main.fodt. Status code: 404
(venv) mblatt@frau-mahlzahn:~/opm-reference-manual/parts$ 
```

**Comment by blattms** [comment]
```
cd ../docker/
(venv) mblatt@frau-mahlzahn:~/opm-reference-manual/docker$ lodocker-run-container main.fodt 
Select a Dockerfile:
1. ubuntu2204-lo7.5.9
2. ubuntu2310-dev
Please enter an integer: 1
INFO:root:git_root: /home/mblatt/opm-reference-manual/venv/lib
access control disabled, clients can connect from any host

(soffice:45): dbind-WARNING **: 09:57:43.583: Couldn't connect to accessibility bus: Failed to connect to socket /run/user/1000/at-spi/bus_0: No such file or directory

(soffice:58): dbind-WARNING **: 09:57:43.976: Couldn't connect to accessibility bus: Failed to connect to socket /run/user/1000/at-spi/bus_0: No such file or directory
```
![image](https://github.com/OPM/opm-reference-manual/assets/1944640/1d4b8bcb-6cce-4156-b527-8d7fbaf69e9a)

**Comment by hakonhagland** [comment]
> $ lodocker-open-file-and-update main.fodt
INFO:werkzeug:172.17.0.1 - - [18/Mar/2024 09:55:27] "POST /open-document-and-update HTTP/1.1" 404 -
Failed to open document main.fodt. Status code: 404

@blattms Did you run `lodocker-start-container` first? `lodocker-open-file-and-update` will not work unless the container web server is running

**Comment by hakonhagland** [comment]
> $ lodocker-open-file-and-update main.fodt

Note that last time we tried `lodocker-open-file-and-update` there was an issue with page numbers, see:

@blattms https://github.com/OPM/opm-reference-manual/pull/108#issuecomment-1927718615

**Comment by blattms** [comment]
> > $ lodocker-open-file-and-update main.fodt
> > INFO:werkzeug:172.17.0.1 - - [18/Mar/2024 09:55:27] "POST /open-document-and-update HTTP/1.1" 404 -
> > Failed to open document main.fodt. Status code: 404
> 
> @blattms Did you run `lodocker-start-container` first? `lodocker-open-file-and-update` will not work unless the container web server is running

It is running. The above message is from the server.  I can use the browser to got to the page and get a 404 error. The error for a non-running server is different (e.g. can’t establish a connection to the server at localhost:8081.)

**Comment by blattms** [comment]
maybe I need to recreate the docker image because something changed?

**Comment by hakonhagland** [comment]
@blattms Yes, right. If the Dockerfile has changed or maybe also if the scripts used by it has changed since you last built the image?

**Comment by blattms** [comment]
Nope. I think it is related to the venv location (mine is normal subdirectory `venv`):
```
docker run -v /home/mblatt/opm-reference-manual/venv/lib/parts:/home/docker-user/parts -v /home/mblatt/opm-reference-manual/venv/lib/fonts:/usr/local/share/fonts:ro --rm -e DISPLAY=:0.0 -e LIBREOFFICE_PORT=2002 -e FLASK_PORT=8080 -e LIBREOFFICE_EXE=libreoffice -e LIBREOFFICE_USERDIR=.config/libreoffice/4 -p 8080:8080 -v /tmp/.X11-unix:/tmp/.X11-unix lo-ubuntu2204
```
Somehow it thinks that files are below venv/lib/-

**Comment by hakonhagland** [comment]
> Somehow it thinks that files are below venv/lib/-

>docker run -v /home/mblatt/opm-reference-manual/venv/lib/parts:/home/docker-user/parts -v /home/mblatt/opm-reference-manual/venv/lib/fonts:/usr/local/share/fonts:ro --rm -e DISPLAY=:0.0 -e LIBREOFFICE_PORT=2002 -e FLASK_PORT=8080 -e LIBREOFFICE_EXE=libreoffice -e LIBREOFFICE_USERDIR=.config/libreoffice/4 -p 8080:8080 -v /tmp/.X11-unix:/tmp/.X11-unix lo-ubuntu2204

@blattms Can you also share the error message you get for running that command?

**Comment by blattms** [comment]
There is no error message. There is an empty directory parts there.

I can fix this easily via removing the directory and putting a symbolic link there to the correct directory:
```
(venv) mblatt@frau-mahlzahn:~/opm-reference-manual/venv/lib$ rmdir parts/
(venv) mblatt@frau-mahlzahn:~/opm-reference-manual/venv/lib$ ln -s ../../parts/
(venv) mblatt@frau-mahlzahn:~/opm-reference-manual/venv/lib$ cd ../../docker; lodocker-start-container
```

**Comment by blattms** [comment]
Maybe, this bogus directory was created when I ran `lodocker-start-container` in the top-level directory? That has to be run in docker subdirectory-

**Comment by blattms** [comment]
I removed venv and recreated it again. Problem persists. 
Removed it again and created one in .venv: Problem persist.

**Comment by blattms** [comment]
The last message is probably the root of the problem when using a venv:
```
$ lodocker-start-container 
Select a Dockerfile:
1. ubuntu2204-lo7.5.9
2. ubuntu2310-dev
Please enter an integer: 1
INFO:root:git_root: /home/mblatt/opm-reference-manual/.venv/lib
```
It thinks the root directory of the git repository is .venv/lib
